File: backend/src/messaging/dto/messaging.dto.ts
export class CreateConversationDto {
  isGroup?: boolean;
  name?: string; // Optional for groups
  description?: string;
  avatarUrl?: string;
  participantIds: string[]; // List of user IDs to include
}

export class UpdateGroupDto {
  name?: string;
  description?: string;
  avatarUrl?: string;
}

export class AddParticipantsDto {
  participantIds: string[];
}

export class SendMessageDto {
  conversationId: string;
  content?: string;
  type?: 'text' | 'image' | 'file' | 'voice';
  fileUrl?: string;
  metadata?: any;
  replyToId?: string;
}

File: goback/models/messaging_dto.go
package models

type StartDirectConversationRequest struct {
	TargetUserID string `json:"targetUserId" validate:"required"`
}

type SendMessageRequest struct {
	ConversationID string          `json:"conversationId"`
	GroupID        string          `json:"groupId"` // For group messages
	Content        string          `json:"content"`
	Type           string          `json:"type"` // text, image, file, voice
	FileURL        string          `json:"fileUrl"`
	Mime           string          `json:"mime"`
	Duration       int             `json:"duration"`
	Size           int             `json:"size"`
	FileURLs       []string        `json:"fileUrls"`
	ReplyToID      string          `json:"replyToId"`
	EntityMentions []EntityMention `json:"entityMentions"`
	IsForwarded    bool            `json:"isForwarded"`
}

type CreateConversationRequest struct {
	IsGroup        bool     `json:"isGroup"`
	Name           string   `json:"name"`
	Description    string   `json:"description"`
	AvatarURL      string   `json:"avatarUrl"`
	ParticipantIDs []string `json:"participantIds"`
}

type AddParticipantsRequest struct {
	ParticipantIDs []string `json:"participantIds"`
}
